Documentation/project-docs/benchmarking.md

```diff
@@ -0,0 +1,102 @@
+> C:\Program Files\dotnet-nightly\shared\Microsoft.NETCore.App\2.1.0-preview1-25919-02
+
+## Shared framework
+You can either decide to use your local self-compiled shared framework package or use the one from the .NET Core 2.1 SDK.
+
+### Using the shared framework from the .NET Core 2.1 SDK
+Follow the instructions described here https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/dogfooding.md#advanced-scenario---using-a-nightly-build-of-microsoftnetcoreapp and skip the last part which calls the `dotnet.exe`.
+
+Now we are going to create a BenchmarkDotNet configuration file to set the path to the .NET Core 2.1 SDK
+
+```
+public class MainConfig : ManualConfig
+{
+    public MainConfig()
+    {
+        Add(Job.Default
+            .With(Runtime.Core)
+            .With(CsProjCoreToolchain.From(new NetCoreAppSettings(
+                targetFrameworkMoniker: "netcoreapp2.1",
+                runtimeFrameworkVersion: "2.1.0-preview1-25919-02", // <-- Adjust version here
+                customDotNetCliPath: @"C:\Program Files\dotnet-nightly\dotnet.exe", // <-- Adjust path here
+                name: "Core 2.1.0-preview")))
+            .WithLaunchCount(1)
+            .WithWarmupCount(1)
+            .WithInvocationCount(10)
+            .WithUnrollFactor(1)
+            .WithTargetCount(3));
+
+        // Add whatever jobs you need
+        Add(DefaultColumnProviders.Instance);
+        Add(MarkdownExporter.GitHub);
+        Add(new ConsoleLogger());
+        Add(new HtmlExporter());
+        Add(MemoryDiagnoser.Default);
+    }
+}
+```
+
+
+
+### Using your self-compiled shared framework
+Follow the instructions described here https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/dogfooding.md#more-advanced-scenario---using-your-local-corefx-build and skip the last part which calls the `dotnet.exe`.
+Make sure to build your local corefx repository in RELEASE mode `.\build -release`! You currently need to have a self-contained application to inject your local shared framework package.
+
+Now we are going to create a BenchmarkDotNet configuration file. Currently there is no easy way to run your BenchmarkDotNet application in a dedicated process, therefore we are using the InProcess switch:
+
+```
+public class MainConfig : ManualConfig
+{
+    public MainConfig()
+    {
+        Add(Job.Default
+            .With(InProcessToolchain.Instance) <-- To run against your local built shared framework (in corefx folder)
+            .WithLaunchCount(1)
+            .WithWarmupCount(1)
+            .WithInvocationCount(1)
+            .WithUnrollFactor(1)
+            .WithTargetCount(3));
+
+        // Add whatever jobs you need
+        Add(DefaultColumnProviders.Instance);
+        Add(MarkdownExporter.GitHub);
+        Add(new ConsoleLogger());
+        Add(new HtmlExporter());
+        Add(MemoryDiagnoser.Default);
+    }
+}
+```
```

Comment:
## Defining your benchmark

See [Benchmark.NET](http://benchmarkdotnet.org/Guides/GettingStarted.htm) documentation -- minimally you need to adorn a public method with the `[Benchmark]` attribute but there are many other ways to customize what is done such as using parameter sets or setup/cleanup methods.

Reply:
Of course, you'll want to bracket just the relevant code in your benchmark, ensure there are sufficient iterations that you minimise noise, as well as leaving the machine otherwise idle while you measure.